pkgs/sops-install-secrets/nixos-test.nix

```diff
@@ -331,6 +331,43 @@ in {
     '';
   };
 
+  template-path-not-external = testers.runNixOSTest {
+    name = "sops-template-path-not-external";
+    nodes.machine = {config, ...}: {
+      imports = [ ../../modules/sops ];
+      sops = {
+        age.keyFile = "/run/age-keys.txt";
+        defaultSopsFile = ./test-assets/secrets.yaml;
+        secrets.test_key = { };
+      };
+
+      # Must run before sops sets up keys.
+      boot.initrd.postDeviceCommands = ''
+        cp -r ${./test-assets/age-keys.txt} /run/age-keys.txt
+        chmod -R 700 /run/age-keys.txt
+      '';
+
+      sops.templates.test_template = {
+        content = ''
+          Test value: ${config.sops.placeholder.test_key}
+        '';
+        # This path is inside of `/run/secrets`, which isn't allowed.
+        path = "/run/secrets/foo-bar";
```

Comment:
If it's inside /run/secrets maybe it should name the template differently rather than doing the symlink? Alternative would be not allow this behavior and make it an eval error.

Reply:
Ooh, I like the "name it differently" idea. I think that'll actually clean up the implementation a bit, as it currently deals carefully (and verbosely) with the "rendered" subdir.

I'll give this a shot later today.